src/main/java/core/basesyntax/UpdateStorageImpl.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.TreeMap;
+
+public class UpdateStorageImpl implements UpdateStorage {
+
+    public void parseFileToStorage(String fileName) throws IOException {
+        List<List<String>> newData = new ReadFromFile().getNewData(fileName);
+        for (List<String> raw : newData) {
+            putLineToStorage(raw);
+        }
+    }
+
+    public void putLineToStorage(List<String> raw) {
+        String operationType = raw.get(0);
+        String fruitType = raw.get(1);
+        int fruitAmount = Integer.parseInt(raw.get(2));
+        LocalDate date = LocalDate.parse(raw.get(3));
+        if (operationType.equals("b") && !Storage.fruitsInStore.containsKey(fruitType)) {
+            return;
+        }
+        if (operationType.equals("b")) {
+            TreeMap<LocalDate, Integer> fruitReminders = new TreeMap<>(Storage
```

Comment:
For variable declaretion better use "Map"

Reply:
done